src/components/pages/Redeem/reducer.ts

```diff
@@ -288,43 +287,7 @@ const updateBassetAmounts = (state: State): State => {
   return state;
 };
 
-const simulateRedeemSingle = ({
-  amountInMasset,
-  bAssets,
-  dataState,
-}: State & { dataState: DataState }): DataState => {
-  const [bAsset] = Object.values(bAssets).filter(b => b.enabled);
-  const { address, amount } = bAsset || {};
-
-  if (!(bAsset && amount && amountInMasset)) {
-    return dataState;
-  }
-
-  const { totalSupply, balance: mAssetBalance, ...mAsset } = dataState.mAsset;
-
-  const {
-    [address]: { balance, totalVault },
-  } = dataState.bAssets;
-
-  return {
-    ...dataState,
-    mAsset: {
-      ...mAsset,
-      totalSupply: totalSupply.sub(amountInMasset),
-      balance: mAssetBalance.sub(amountInMasset),
-    },
-    bAssets: {
-      ...dataState.bAssets,
-      [address]: {
-        ...dataState.bAssets[address],
-        balance: balance.add(amount),
-        totalVault: totalVault.sub(amount),
-      },
-    },
-  };
-};
-
-const simulateRedeemMasset = ({
+const simulateRedemption = ({
```

Comment:
Turns out the `simulateRedeemSingle` logic isn't necessary, given that all of the inputs have an `enabled` flag, and their amounts are set correctly.